src/databricks/labs/blueprint/installation.py

```diff
@@ -481,7 +482,11 @@ def _marshal(self, type_ref: type, path: list[str], inst: Any) -> tuple[Any, boo
         if dataclasses.is_dataclass(type_ref):
             return self._marshal_dataclass(type_ref, path, inst)
         if type_ref == list:
-            return self._marshal_list(type_ref, path, inst)
+            return self._marshal_raw_list(path, inst)
+        if type_ref == dict:
+            return self._marshal_raw_dict(path, inst)
+        if type_ref in (object, Any):
+            return self._marshal(type(inst), path, inst)
```

Comment:
I'm not so sure about this, it looks like the caller isn't providing the correct `type_ref`?

The intent of that parameter is that the caller should know the type of what it's trying to marshal and declare it up-front. With this change I'm not sure what the purpose of the parameter is at all?

Reply:
There are situations where the `type_ref` is not specific, because it's not specified by the caller, notably when serializing/deserializing nested structures. Generally speaking, the code will produce better results if the result types are accurately specified, but if not, it will return a list or dict, which can easily be converted by the caller, instead of crashing.